contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -92,8 +245,127 @@ contract ProtocolShareReserve is Ownable2StepUpgradeable, ExponentialNoError, Re
+        address[] memory markets = IPrime(prime).allMarkets();
+        for (uint i = 0; i < markets.length; ++i) {
+            address market = markets[i];
+            IPrime(prime).accrueInterest(market);
+            _releaseFund(CORE_POOL_COMPTROLLER, _getUnderlying(market));
+        }
+    }
+
+    /**
+     * @dev Fetches the list of prime markets and then accrues interest
+     * to prime for each market
+     */
+    function _accruePrimeInterest() internal {
+        address[] memory markets = IPrime(prime).allMarkets();
+        for (uint i = 0; i < markets.length; ++i) {
+            address market = markets[i];
+            IPrime(prime).accrueInterest(market);
+        }
+    }
+
+    function _releaseFund(address comptroller, address asset) internal {
+        uint256 totalSchemas = uint256(type(Schema).max) + 1;
+        uint[] memory schemaBalances = new uint[](totalSchemas);
+        uint256 totalBalance;
+
+        for (uint schemaValue = 0; schemaValue <= totalSchemas - 1; ++schemaValue) {
+            schemaBalances[schemaValue] = assetsReserves[comptroller][asset][Schema(schemaValue)];
+            totalBalance = schemaBalances[schemaValue];
+        }
+
+        if (totalBalance == 0) {
+            return;
+        }
+
+        uint[] memory totalTransferAmounts = new uint[](totalSchemas);
+
+        for (uint i = 0; i < distributionTargets.length; ++i) {
+            DistributionConfig memory _config = distributionTargets[i];
+
+            uint256 transferAmount = schemaBalances[uint256(_config.schema)] * _config.percentage / MAX_PERCENT;
+            totalTransferAmounts[uint256(_config.schema)] += transferAmount;
+
+            IERC20Upgradeable(asset).safeTransfer(_config.destination, transferAmount);
+            IIncomeDestination(_config.destination).updateAssetsState(comptroller, asset);
+
+            emit AssetReleased(_config.destination, asset, _config.schema, _config.percentage, transferAmount);
+        }
+
+        uint[] memory newSchemaBalances = new uint[](totalSchemas);
+        for (uint schemaValue = 0; schemaValue <= totalSchemas - 1; ++schemaValue) {
+            newSchemaBalances[schemaValue] = schemaBalances[schemaValue] - totalTransferAmounts[schemaValue];
+            assetsReserves[comptroller][asset][Schema(schemaValue)] = newSchemaBalances[schemaValue];
+            totalAssetReserve[asset] = totalAssetReserve[asset] - totalTransferAmounts[schemaValue];
+
+            emit ReservesUpdated(comptroller, asset, Schema(schemaValue), schemaBalances[schemaValue], newSchemaBalances[schemaValue]);
+        }
+    }
+
+    function _getUnderlying(address vToken) internal view returns (address) {
+        if (vToken == vBNB) {
+            return WBNB;
+        } else {
+            return IVToken(vToken).underlying();
+        }
+    }
+
+    function getSchema(address comptroller, address asset, IncomeType incomeType) internal returns (Schema schema) {
+        schema = Schema.SPREAD_PRIME_CORE;
```

Comment:
I think it's the other way, first we assign `DEFAULT` and if the `if` statement is true we assign `SPREAD_PRIME_CORE` schema